File: restdocs/src/main/java/com/restdocs/sample/SampleController.java
package com.restdocs.sample;

import lombok.Getter;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("sample")
public class SampleController {

    @RequestMapping(method = RequestMethod.GET)
    @ResponseStatus(value = HttpStatus.OK)
    public Sample getSample() {
        return new Sample();
    }

    @Getter
    public static class Sample {
        private String name = "yun";
        private int age = 27;
    }

}

File: boot3/src/main/kotlin/org/example/boot3/SampleController.kt
package org.example.boot3

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/sample")
class SampleController {

    @GetMapping("/hello")
    fun hello(): SampleResponse {

        return SampleResponse(
            name = "Hello, World!"
        )
    }
}

data class SampleResponse(
    val name: String
)